Tasks/ArchiveFiles/archivefiles.ts

```diff
@@ -0,0 +1,273 @@
+    // behavior resulting in archives that are correct but are renamed
+    // e.g. test.tz2 has test inside of it instead of test.tar (7z decides to rename it)
+    // e.g. test_tgz has an outer name of test_tgz.gz (7z decides to add the .gz to it).
+
+    //non standard name
+
+    //foo.tar.anything --> foo.tar
+    if (lowerArchiveName.lastIndexOf('.tar.') > 0){
+        return archiveName.substring(0, lowerArchiveName.lastIndexOf('.tar.') + 4);
+    }
+    // foo.anything --> foo.tar
+    else if (lowerArchiveName.lastIndexOf('.') > 0){
+        return archiveName.substring(0, lowerArchiveName.lastIndexOf('.'))+'.tar';
+    }
+    // foo --> foo.tar
+    return lowerArchiveName + '.tar';
+}
+
+//ensure output folder exists
+var destinationFolder = path.dirname(archiveFile);
+tl.debug("Checking for archive destination folder:"+destinationFolder);
+if(!tl.exist(destinationFolder)){
+    tl.debug("Destination folder does not exist, creating:"+destinationFolder);
+    tl.mkdirP(destinationFolder);
+}
+
+if(win) { // windows only
+    if(archiveType == "default"){ //default is zip format
+        sevenZipArchive(archiveFile, "zip", files);
+    } else if(archiveType == "tar"){
+        var tarCompression: string = tl.getInput('tarCompression', true);
+        if(tarCompression == "none"){
+            sevenZipArchive(archiveFile, archiveType, files);        
+        } else {
+            var tarFile = computeTarName(archiveFile);
+            var tarExists = tl.exist(tarFile);
+            try {
+                if(tarExists){
+                    console.log('Intermediate tar: '+tarFile +' already exists.  Attempting to add files to it.');
+                }
+                
+                // this file could already exist, but not checking because by default files will be added to it
+                // create the tar file
+                sevenZipArchive(tarFile, archiveType, files);
+                // compress the tar file
+                var sevenZipCompressionFlag = tarCompression;
+                if(tarCompression == 'gz'){
+                    sevenZipCompressionFlag = 'gzip';
+                } else if (tarCompression == 'bz2'){
+                    sevenZipCompressionFlag = 'bzip2';
+                }
+                sevenZipArchive(archiveFile, sevenZipCompressionFlag, [tarFile]);
+            } finally {
+                // delete the tar file if we created it.
+                if(!tarExists){
+                    fs.unlinkSync(tarFile);
+                }
+            }
+        }
+    } else {
+        sevenZipArchive(archiveFile, archiveType, files);
+    }
+} else { // not windows
+    if (archiveType == "default"){ //default is zip format
+        zipArchive(archiveFile, files);
+    } else if(archiveType == "tar"){
+        var tarCompression: string = tl.getInput('tarCompression', true);
+        var tarCompressionFlag;
+        if(tarCompression == "none"){
```

Comment:
If you change "none" to "None" in task.json, this line needs to change too (see comment in task.json).